readthedocs/rtd_tests/tests/test_doc_builder.py

```diff
@@ -488,6 +488,42 @@ def test_write_js_data_docs_dir(self, checkout_path, run, generate_rtd_data):
             mkdocs_config=mock.ANY,
         )
 
+    @patch('readthedocs.doc_builder.backends.mkdocs.BaseMkdocs.generate_rtd_data')
+    @patch('readthedocs.projects.models.Project.checkout_path')
+    def test_write_js_data_on_invalid_docs_dir(self, checkout_path, generate_rtd_data):
+        tmpdir = tempfile.mkdtemp()
+        os.mkdir(os.path.join(tmpdir, 'docs'))
+        yaml_file = os.path.join(tmpdir, 'mkdocs.yml')
+        yaml.safe_dump(
+            {
+                'site_name': 'mkdocs',
+                'google_analytics': ['UA-1234-5', 'mkdocs.org'],
+                'docs_dir': 'invalid_docs_dir',
+                'extra_css': [
+                    'http://readthedocs.org/static/css/badge_only.css'
+                ],
+                'extra_javascript': ['readthedocs-data.js'],
+            },
+            open(yaml_file, 'w'),
+        )
+        checkout_path.return_value = tmpdir
+
+        python_env = Virtualenv(
+            version=self.version,
+            build_env=self.build_env,
+            config=None,
+        )
+        self.searchbuilder = MkdocsHTML(
+            build_env=self.build_env,
+            python_env=python_env,
+        )
+        with self.assertRaises(MkDocsYAMLParseError):
+            self.searchbuilder.append_conf()
+            generate_rtd_data.assert_called_with(
```

Comment:
I guess we don't need this assert

Reply:
@stsewd then how should we check if invalid `docs_path` raises an error or not?

Reply:
Yes, we only need that check. The second assert wouldn't happen anyway.

Reply:
yes that's True. :+1:  updated the PR.